website_docs/configuring_automatic_instrumentation.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Configuring automatic instrumentation
+weight: 5
+---
+
+Automatic instrumentation in ruby is done via instrumentation packages, and most commonly, the `opentelemetry-instrumentation-all` package. These are alled Instrumentation Libraries.
+
+For example, if you are using Rails and enable instrumentation, your running Rails app will automatically generate telemetry data for inbound requests to your controllers.
+
+### Configuring all instrumentation libraries
+
+The recommended way to use instrumentation libraries is to simply use the `opentelemetry-instrumentation-all` package:
+
+```console
+gem 'opentelemetry-sdk'
+gem 'opentelemetry-exporter-otlp'
+gem 'opentelemetry-instrumentation-all'
+```
+
+and configure it early in your application lifecycle (for Rails apps, this would be in the Rails initializer):
+
+```ruby
+# config/initializers/opentelemetry.rb
+require 'opentelemetry/sdk'
+require 'opentelemetry/exporter/otlp'
+require 'opentelemetry/instrumentation/all'
+OpenTelemetry::SDK.configure do |c|
+  c.service_name = '<YOUR_SERVICE_NAME>'
+  c.use_all() # enables all instrumentation!
+end
+```
+
+This will install all instrumentation libraries and enable the ones that match up to libraries you're using in your app.
+
+### Configuring specific instrumentation libraries
+
+If you prefer more selectively installing and using only specific instrumentation libraries, you can do that too. For example, here's how to use only `Sinatra` and `Faraday`, with `Farady` being configured with an additional configuration parameter.
+
+First, install the specific instrumentation libraries you know you want to use:
+
+```console
+gem install opentelemetry-instrumentation-sinatra
+gem install opentelemetry-instrumentation-faraday
+```
+
+The configure them:
+
+
+```ruby
+require 'opentelemetry/sdk'
+
+# install all compatible instrumentation with default configuration
+OpenTelemetry::SDK.configure do |c|
+  c.use 'OpenTelemetry::Instrumentation::Sinatra'
+  c.use 'OpenTelemetry::Instrumentation::Faraday', { opt: 'value' }
+end
+```
+
+### Overriding configuration for specific instrumentation libraries
+
+If you are enabling all instrumentation but want to override the configuration for a specific one, call `use_all` with a configuration parameter:
+
+```ruby
+require 'opentelemetry/sdk'
+
+# install all compatible instrumentation with per instrumentation configuration overrides
+OpenTelemetry::SDK.configure do |c|
+  c.use_all('OpenTelemetry::Instrumentation::SomeInstrumentation' => { opt: 'value' })
```

Comment:
I think providing a concrete example would be best. 

A common case we have heard from our users is how to disable instrumentation for Redis because of how many spans it generates. Another is how to enable DB obfuscation. How would you feel about using something like that as the bases for the example here?

Reply:
Great feedback - done and done!